File: src/main/kotlin/net/cjsah/bot/console/command/CommandException.java
package net.cjsah.bot.console.command;

public class CommandException {

    public static final CException UNKNOWN_COMMAND = new UnknownCommandException();
    public static final CException PARAMETER_ERROR = new ParameterErrorException();

    static class CException extends Exception {
        protected CException(String e) {
            super(e);
        }
    }

    static class UnknownCommandException extends CException {
        private UnknownCommandException() {
            super("Unknown Command!");
        }
    }

    static class ParameterErrorException extends CException {
        private ParameterErrorException() {
            super("Parameter Error!");
        }
    }

}

File: src/main/kotlin/net/cjsah/bot/console/command/CommandException.kt
package net.cjsah.bot.console.command

@Deprecated("")
object CommandException {

    val UNKNOWN_COMMAND: CException = UnknownCommandException()
    val PARAMETER_ERROR: CException = ParameterErrorException()

    open class CException protected constructor(e: String) : Exception(e)

    internal class UnknownCommandException : CException("Unknown Command!")

    internal class ParameterErrorException : CException("Parameter Error!")

}